src/NzbDrone.Core.Test/HistoryTests/HistorySpecificationFixture.cs

```diff
@@ -0,0 +1,273 @@
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.CustomFormats;
+using NzbDrone.Core.DecisionEngine;
+using NzbDrone.Core.DecisionEngine.Specifications.RssSync;
+using NzbDrone.Core.History;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.Profiles;
+using NzbDrone.Core.Profiles.Qualities;
+using NzbDrone.Core.Qualities;
+using NzbDrone.Core.Test.Framework;
+using NzbDrone.Core.Test.Qualities;
+using NzbDrone.Core.Tv;
+
+namespace NzbDrone.Core.Test.HistoryTests
+{
+    [TestFixture]
+    public class HistorySpecificationFixture : CoreTest<HistorySpecification>
+    {
+        private RemoteEpisode _remoteEpisode;
+        private Series _series;
+        private Episode _episode;
+        private QualityModel _quality;
+        private QualityProfile _profile;
+
+        [SetUp]
+        public void Setup()
+        {
+            _series = Builder<Series>.CreateNew()
+                .With(s => s.Id = 1)
+                .Build();
+
+            _episode = Builder<Episode>.CreateNew()
+                .With(e => e.Id = 1)
+                .With(e => e.SeriesId = _series.Id)
+                .Build();
+
+            _quality = new QualityModel(Quality.HDTV720p);
+            _series.QualityProfile = _profile;
+
+            _remoteEpisode = Builder<RemoteEpisode>.CreateNew()
+                .With(r => r.Series = _series)
+                .With(r => r.Episodes = new List<Episode> { _episode })
+                .With(r => r.ParsedEpisodeInfo = new ParsedEpisodeInfo { Quality = _quality })
+                .With(r => r.CustomFormats = new List<CustomFormat>())
+                .With(r => r.CustomFormatScore = 0)
+                .Build();
+
+            _profile = new QualityProfile
+            {
+                Cutoff = Quality.HDTV720p.Id,
+                Items = QualityFixture.GetDefaultQualities(),
+                Name = "Test"
+            };
+        }
+
+        private void SetupHistoryServiceMock(List<EpisodeHistory> history)
+        {
+            Mocker.GetMock<IHistoryService>()
+                .Setup(s => s.FindByEpisodeId(It.IsAny<int>()))
+                .Returns(history);
+        }
+
+        private void SetupCdh(bool cdhEnabled)
+        {
+            Mocker.GetMock<IConfigService>()
+                .Setup(s => s.EnableCompletedDownloadHandling)
+                .Returns(cdhEnabled);
+        }
+
+        private List<EpisodeHistory> GivenFileHistory(DateTime date, QualityModel quality, EpisodeHistoryEventType eventType = EpisodeHistoryEventType.SeriesFolderImported)
+        {
+            return new List<EpisodeHistory>
+            {
+                new()
+                {
+                    Date = date,
+                    Quality = quality,
+                    EventType = eventType
+                }
+            };
+        }
+
+        private RemoteEpisode MockUpOrDownGrade(int customFormatScoreHistory, int customFormatScoreNew, QualityModel newQuality, int cutoff, bool hasFile = false)
```

Comment:
This is doing too much and should be split into  `GivenProfile` that does the profile + mock and then a separate function to create the `RemoteEpisode`